debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -282,8 +282,10 @@ public void getChangesForTables(String databaseName, SqlServerChangeTable[] chan
         prepareQuery(queries, preparers, consumer);
     }
 
-    private Lsn getFromLsn(String databaseName, SqlServerChangeTable changeTable, Lsn intervalFromLsn) throws SQLException {
-        Lsn fromLsn = changeTable.getStartLsn().compareTo(intervalFromLsn) > 0 ? changeTable.getStartLsn() : intervalFromLsn;
+    private Lsn getFromLsn(String databaseName, SqlServerChangeTable changeTable, Lsn intervalFromLsn, Lsn intervalToLsn) throws SQLException {
+        Lsn fromLsn = changeTable.getStartLsn().compareTo(intervalFromLsn) > 0 && changeTable.getStartLsn().compareTo(intervalToLsn) < 0
+                ? changeTable.getStartLsn()
+                : intervalFromLsn;
         return fromLsn.getBinary() != null ? fromLsn : getMinLsn(databaseName, changeTable.getCaptureInstance());
```

Comment:
sometimes (at least in tests) it's still `null`, but that's ok for `fromLsn`

Reply:
I believe the `fromLsn` may be null for a new connector that doesn't have any stored offsets. But then it should probably default to the value of `getMinLsn()`.